src/app.ts

```diff
@@ -67,24 +67,11 @@ function withSpidAuth(
     passport.authenticate("spid", async (err, user) => {
       if (err) {
         log.error("Error in SPID authentication: %s", err);
-        const parser = new DOMParser();
-        const xmlDoc = parser.parseFromString(err.statusXml || "", "text/xml");
-        if (!!xmlDoc) {
-          const errorElement = xmlDoc.getElementsByTagName("StatusMessage");
-          if (errorElement.length > 0) {
-            const indexString = "ErrorCode nr";
-            const errorString = errorElement[0].textContent || "";
-            const errorCode = errorString.slice(
-              errorString.indexOf(indexString) + indexString.length
-            );
-
-            return res.redirect(
-              clientErrorRedirectionUrl +
-                (errorCode.length ? `?errorCode=${errorCode}` : "")
-            );
-          }
-        }
-        return res.redirect(clientErrorRedirectionUrl);
+        const errorCode = getErrorCodeFromResponse(err.statusXml);
```

Comment:
is `statusXml` always defined?

Reply:
I took care about this evaluation and enhanced controls in a0a719c